File: tower-of-hanoi.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

void TOH(int n, char A, char B, char C)
{
    if (n == 1)
    {
        cout << "Move 1 from " << A << " to " << C << endl;
        return;
    }
    TOH(n - 1, A, C, B);
    cout << "Move " << n << " from " << A << " to " << C << endl;
    TOH(n - 1, B, A, C);
}

int main()
{
    //region READ FROM FILE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    //end region
    TOH(5, 'A', 'C', 'B');
    return 0;
}

File: python/gfg/3.Recurrsion/Tower_of_hanoi.py
def TOH(n, a, b, c):
    if n == 1:
        print(f"Move 1 from {a} to {b}")
        return
    TOH(n - 1, a, c, b)
    print(f"Move {n} from {a} to {c}")
    TOH(n - 1, b, a, c)


n = int(input())
TOH(n, 'A', 'B', 'C')